assembl/models/social_auth.py

```diff
@@ -135,6 +135,15 @@ class SocialAuthAccount(
     def provider(self):
         return self.identity_provider.provider_type
 
+    @property
+    def provider_with_idp(self):
+        provider = self.provider
+        if provider == 'saml':
+            idp_name = self.uid.split(':')[0]
```

Comment:
The IDP name is used all over the place, but this new method is only used in Discussion.check_authorized_email